controllers/policystatus/policy_status_controller.go

```diff
@@ -0,0 +1,140 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package policystatus
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	policiesv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+	"open-cluster-management.io/governance-policy-propagator/controllers/common"
+	"open-cluster-management.io/governance-policy-propagator/controllers/propagator"
+)
+
+const ControllerName string = "policy-status"
+
+var log = ctrl.Log.WithName(ControllerName)
+
+//+kubebuilder:rbac:groups=policy.open-cluster-management.io,resources=policies,verbs=get;list;watch
+//+kubebuilder:rbac:groups=policy.open-cluster-management.io,resources=policies/status,verbs=get;update;patch
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *PolicyStatusReconciler) SetupWithManager(mgr ctrl.Manager, additionalSources ...source.Source) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		WithOptions(controller.Options{MaxConcurrentReconciles: int(r.MaxConcurrentReconciles)}).
+		Named(ControllerName).
+		For(
+			&policiesv1.Policy{},
+			builder.WithPredicates(policyStatusPredicate(mgr.GetClient())),
+		).
+		Complete(r)
+}
+
+// blank assignment to verify that PolicyStatusReconciler implements reconcile.Reconciler
+var _ reconcile.Reconciler = &PolicyStatusReconciler{}
+
+// PolicyStatusReconciler handles replicated policy status updates and updates the root policy status.
+type PolicyStatusReconciler struct {
+	client.Client
+	MaxConcurrentReconciles uint
+	// Use a shared lock with the main policy controller to avoid conflicting updates.
+	RootPolicyLocks *common.PoliciesLock
+	Scheme          *runtime.Scheme
+}
+
+// Reconcile handles a replicated policy status update and then updates the root policy status. It does not handle
+// creations or deletions of replicated policies. Those are handled by the main policy controller. This Reconcile
+// may have a lot retries if several clusters are updating the same policy at the same time due to a race condition
+// where the root policy status update hasn't been updated in the cache.
+func (r *PolicyStatusReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	log := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	log.Info("Reconciling the root policy status")
```

Comment:
I'm thinking this should be `V(1)` - since it comes from every replicated policy, it could be very noisy in a busy cluster. Right?

Reply:
Yeah, you're right. I'll change it.